src/parser/reference-parser.ts

```diff
@@ -147,6 +162,18 @@ export function parseColumnName(column: AnyColumn<any, any>): ColumnNode {
   return ColumnNode.create(column as string)
 }
 
+export function parseOrderedColumnName(column: string): OperationNode {
+  const ORDER_SEPARATOR = ' '
+
+  if (column.includes(ORDER_SEPARATOR)) {
+    const [columnName, order] = column.split(ORDER_SEPARATOR).map(trim)
+
+    return parseOrderBy(columnName, order as any)
```

Comment:
You could check the `order` argument instead of casting to any:

```ts
function isOrder(order: string): order is 'asc' | 'desc'
```

Reply:
Good point, Added the guard to `order-by-parser` next to `OrderByDirection` type.